cmd/openstack-populator/openstack-populator.go

```diff
@@ -11,136 +11,174 @@ import (
 	libclient "github.com/konveyor/forklift-controller/pkg/lib/client/openstack"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
+	dto "github.com/prometheus/client_model/go"
 	"k8s.io/klog/v2"
 )
 
-func main() {
-	var (
-		identityEndpoint string
-		imageID          string
-		crNamespace      string
-		crName           string
-		secretName       string
-
-		volumePath string
-	)
+type AppConfig struct {
+	identityEndpoint string
+	imageID          string
+	crNamespace      string
+	crName           string
+	secretName       string
+	ownerUID         string
+	pvcSize          int64
+	volumePath       string
+}
 
+func main() {
+	config := parseFlags()
 	klog.InitFlags(nil)
+	initializeMetrics()
+	populate(config)
+}
 
-	// Main arg
-	flag.StringVar(&identityEndpoint, "endpoint", "", "endpoint URL (https://openstack.example.com:5000/v2.0)")
-	flag.StringVar(&secretName, "secret-name", "", "secret containing OpenStack credentials")
-
-	flag.StringVar(&imageID, "image-id", "", "Openstack image ID")
-	flag.StringVar(&volumePath, "volume-path", "", "Path to populate")
-	flag.StringVar(&crName, "cr-name", "", "Custom Resource instance name")
-	flag.StringVar(&crNamespace, "cr-namespace", "", "Custom Resource instance namespace")
-
+func parseFlags() *AppConfig {
+	config := &AppConfig{}
+	flag.StringVar(&config.identityEndpoint, "endpoint", "", "endpoint URL (https://openstack.example.com:5000/v2.0)")
+	flag.StringVar(&config.secretName, "secret-name", "", "secret containing OpenStack credentials")
+	flag.StringVar(&config.imageID, "image-id", "", "Openstack image ID")
+	flag.StringVar(&config.volumePath, "volume-path", "", "Path to populate")
+	flag.StringVar(&config.crName, "cr-name", "", "Custom Resource instance name")
+	flag.StringVar(&config.crNamespace, "cr-namespace", "", "Custom Resource instance namespace")
+	flag.StringVar(&config.ownerUID, "owner-uid", "", "Owner UID (usually PVC UID)")
+	flag.Int64Var(&config.pvcSize, "pvc-size", 0, "Size of pvc (in bytes)")
```

Comment:
resolved the thread I commented on this before but I see that we still default this to 0 which is a bit unexpected, shouldn't we panic if we don't get `pvc-size`?

Reply:
the `flag` package doesn't seem to provide required flags, I can add a manual check after parsing

Reply:
looks good